test/integration/envtest/shootmaintenance/maintenance_control_test.go

```diff
@@ -0,0 +1,267 @@
+			err = StartShootMaintenance(ctx, testClient, shoot)
+			Expect(err).To(BeNil())
+
+			err = WaitForExpectedKubernetesVersionMaintenance(ctx, Logger, testClient, shoot, testKubernetesVersionLowPatchLowMinor.Version, false, time.Now().Add(time.Second*10))
+			Expect(err).To(BeNil())
+
+		}, waitForCreateDeleteTimeout.Seconds())
+
+		It("Kubernetes version should be updated: auto update enabled", func() {
+			// set test specific shoot settings
+			patch := client.MergeFrom(shoot.DeepCopy())
+			shoot.Spec.Maintenance.AutoUpdate.KubernetesVersion = true
+			Expect(testClient.Patch(ctx, shoot, patch))
+
+			err = StartShootMaintenance(ctx, testClient, shoot)
+			Expect(err).To(BeNil())
+
+			err = WaitForExpectedKubernetesVersionMaintenance(ctx, Logger, testClient, shoot, testKubernetesVersionHighestPatchLowMinor.Version, true, time.Now().Add(time.Second*20))
+			Expect(err).To(BeNil())
+		}, waitForCreateDeleteTimeout.Seconds())
+
+		It("Kubernetes version should be updated: force update patch version", func() {
+			defer func() {
+				// make sure to remove expiration date from cloud profile after test
+				err = PatchCloudProfileForKubernetesVersionMaintenance(ctx, testClient, shoot.Spec.CloudProfileName, testKubernetesVersionLowPatchLowMinor.Version, nil, &deprecatedClassification)
+				Expect(err).To(BeNil())
+				Logger.Infof("Cleaned expiration date on kubernetes version from Cloud Profile '%s'", cloudProfile.Name)
+			}()
+
+			// expire the Shoot's Kubernetes version
+			err = PatchCloudProfileForKubernetesVersionMaintenance(ctx, testClient, shoot.Spec.CloudProfileName, testKubernetesVersionLowPatchLowMinor.Version, &expirationDateInThePast, &deprecatedClassification)
+			Expect(err).To(BeNil())
+
+			// give controller caches time to sync
+			time.Sleep(10 * time.Second)
+
+			err = StartShootMaintenance(ctx, testClient, shoot)
+			Expect(err).To(BeNil())
+
+			err = WaitForExpectedKubernetesVersionMaintenance(ctx, Logger, testClient, shoot, testKubernetesVersionHighestPatchLowMinor.Version, true, time.Now().Add(time.Second*20))
+			Expect(err).To(BeNil())
+		}, waitForCreateDeleteTimeout.Seconds())
+
+		It("Kubernetes version should be updated: force update minor version", func() {
+			defer func() {
+				// make sure to remove expiration date from cloud profile after test
+				err = PatchCloudProfileForKubernetesVersionMaintenance(ctx, testClient, shoot.Spec.CloudProfileName, testKubernetesVersionHighestPatchLowMinor.Version, &expirationDateInThePast, &deprecatedClassification)
+				Expect(err).To(BeNil())
+				Logger.Infof("Cleaned expiration date on kubernetes version from Cloud Profile '%s'", cloudProfile.Name)
+			}()
+
+			// set the shoots Kubernetes version to be the highest patch version of the minor version
+			patch := client.MergeFrom(shoot.DeepCopy())
+			shoot.Spec.Kubernetes.Version = testKubernetesVersionHighestPatchLowMinor.Version
+			framework.ExpectNoError(testClient.Patch(ctx, shoot, patch))
```

Comment:
For consistency with the rest of the file

```suggestion
			err = testClient.Patch(ctx, shoot, patch)
			Expect(err).To(BeNil())
```

(I know that this is inherited from the old version of the test)